js/sites.js

```diff
@@ -265,20 +266,26 @@
      */
     refreshThumbnail() {
       gNewTab.sendToBrowser("NewTab:PageThumbs", {
-        url: this.url
+        link: this.link
       });
     },
 
-    _getURI(message) {
+    /**
+     * Render the correct thumbnail for the site.
+     *
+     * @param {Object} aData Contains enhanced links and the URI generated from the
+     *        current URL.
+     */
+    _getURI(aData) {
       // Only enhance tiles if that feature is turned on
-      let link = /*message.enhanced && DirectoryLinksProvider.getEnhancedLink(this.link) || */ this.link;
+      let link = aData.enhanced && this.link;
 
       let thumbnail = this._querySelector(".newtab-thumbnail");
       if (link.bgColor) {
```

Comment:
hmm.. confused here... the above makes it look like link is a boolean, but then we are checking for an attribute "bgColor"